x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/transport/netty4/SimpleSecurityNetty4TransportTests.java

```diff
@@ -205,31 +204,28 @@ public void testRenegotiation() throws Exception {
             stream.writeInt(-1);
             stream.flush();
 
-            socket.startHandshake();
             CountDownLatch renegotiationLatch = new CountDownLatch(1);
             HandshakeCompletedListener secondListener = event -> renegotiationLatch.countDown();
             socket.addHandshakeCompletedListener(secondListener);
-
-            AtomicReference<Exception> error = new AtomicReference<>();
-            CountDownLatch catchReadErrorsLatch = new CountDownLatch(1);
-            Thread renegotiationThread = new Thread(() -> {
-                try {
-                    socket.setSoTimeout(50);
-                    socket.getInputStream().read();
-                } catch (SocketTimeoutException e) {
-                    // Ignore. We expect a timeout.
-                } catch (IOException e) {
-                    error.set(e);
-                } finally {
-                    catchReadErrorsLatch.countDown();
+            socket.startHandshake();
+            AtomicBoolean stopped = new AtomicBoolean(false);
+            socket.setSoTimeout(50);
```

Comment:
Let's reduce the timeout now. Maybe 5-10 ms since we are retrying. No reason to require this test to take 50 ms when it will normally take much less (now that we are retrying).